events/events_handler.go

```diff
@@ -0,0 +1,309 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package events
+
+import (
+	"sort"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/cadvisor/utils/oomparser"
+)
+
+// EventManager is implemented by Events. It provides three main ways to monitor
+// events
+type EventManager interface {
+	// Watch() provides a channel to which EventManager types are streamed as
+	// they are detected, but before being fed to the channel may be screened
+	// by parameter defined in the passed *Request object
+	Watch(eventChannel chan *Event, request *Request)
```

Comment:
Hmmm I was seeing this more like there is one "Events" in the system and everyone uses that instead of there being more than one. I also viewed the internal as private to it. What else did you have in mind?